src/sql/workbench/contrib/connection/browser/connectionStatus.ts

```diff
@@ -59,16 +64,38 @@ export class ConnectionStatusbarItem extends Disposable implements IWorkbenchCon
 	// Update the connection status shown in the bar
 	private _updateStatus(): void {
 		let activeConnection = TaskUtilities.getCurrentGlobalConnection(this._objectExplorerService, this._connectionManagementService, this._editorService, this._logService);
+		let id = undefined;
+		if (this._editorService.activeEditor) {
+			// USE ACTIVE EDITOR INFO AS THE ID WILL BE DIFFERENT FOR EDITOR CONNECTION.
+			let newInfo = this._connectionManagementService.getConnectionInfo(this._editorService.activeEditor.resource.toString());
+			id = newInfo?.serverConnectionId;
+		}
 		if (activeConnection) {
-			this._setConnectionText(activeConnection);
+			this._setConnectionText(activeConnection, id)
 			this.show();
-		} else {
+		}
+		else {
 			this.hide();
 		}
 	}
 
+	private _refreshIDStatus(uri: string, id: string | undefined): void {
+		let activeConnection = TaskUtilities.getCurrentGlobalConnection(this._objectExplorerService, this._connectionManagementService, this._editorService, this._logService);
```

Comment:
nit: I would suggest pulling the shared activeConnection code from here and _updateStatus. We want those to always used the same logic for the "active" connection, and right now it'd be easy to make a change in one and forget to update the other. 

(then creating a single function for the ID logic as well would be good for the same reason)